DSCResources/MSFT_xIisMimeTypeMapping/MSFT_xIisMimeTypeMapping.psm1

```diff
@@ -0,0 +1,168 @@
+        $getTargetResourceResult = @{Extension = $mt.fileExtension
+                                    MimeType = $mt.mimeType}
+    }
+    
+	return $getTargetResourceResult
+}
+
+######################################################################################
+# The Set-TargetResource cmdlet.
+# This function will add or remove a MIME type mapping
+######################################################################################
+function Set-TargetResource
+{
+	param
+	(	
+        [ValidateSet("Present", "Absent")]
+        [Parameter(Mandatory)]
+        [string]$Ensure = "Present",
+        	
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$Extension,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [String]$MimeType
+    )
+
+        CheckIISPoshModule
+
+        [string]$psPathRoot = "MACHINE/WEBROOT/APPHOST"
+        [string]$sectionNode = "system.webServer/staticContent"
+
+        $mt = GetMapping $Extension $MimeType 
+
+        if ($mt -eq $null -and $Ensure -eq "Present")
+        {
+            # add the MimeType            
+            Add-WebConfigurationProperty -pspath $psPathRoot  -filter $sectionNode -name "." -value @{fileExtension="$Extension";mimeType="$MimeType"}
+            Write-Verbose($LocalizedData.AddingType -f $MimeType,$Extension);
+        }
+        elseif ($mt -ne $null -and $Ensure -eq "Absent")
+        {
+            # remove the MimeType                      
+            Remove-WebConfigurationProperty -pspath $psPathRoot -filter $sectionNode -name "." -AtElement @{fileExtension="$Extension"}
+            Write-Verbose($LocalizedData.RemovingType -f $MimeType,$Extension);
+        }
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test if the given MIME type mapping has the desired state, Present or Absent
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+	param
+	(	
+		[Parameter(Mandatory)]
+        [ValidateSet("Present", "Absent")]
+        [string]$Ensure = "Present",
+	
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+		[String]$Extension,
+
+		[Parameter(Mandatory)]
+		[ValidateNotNullOrEmpty()]
+        [String]$MimeType
+	)
+
+    [bool]$DesiredConfigurationMatch = $true;
+	
+    CheckIISPoshModule
+
+    $mt = GetMapping $Extension $MimeType 
+
+    if (($mt -eq $null -and $Ensure -eq "Present") -or ($mt -ne $null -and $Ensure -eq "Absent"))
+    {
+        $DesiredConfigurationMatch = $false;
+    }
+    elseif ($mt -ne $null -and $Ensure -eq "Present")
+    {
+        # Already there 
+        Write-Verbose($LocalizedData.TypeExists -f $MimeType,$Extension);
+    }
+    elseif ($mt -eq $null -and $Ensure -eq "Absent")
+    {
+        # TypeNotPresent
+        Write-Verbose($LocalizedData.TypeNotPresent -f $MimeType,$Extension);
```

Comment:
Are you talking about the `Write-Verbose`? - No not really needed, I can take them out.